openhands/server/routes/public.py

```diff
@@ -104,3 +105,24 @@ async def get_security_analyzers():
         list: A sorted list of security analyzer names.
     """
     return sorted(SecurityAnalyzers.keys())
+
+
+@app.get('/config')
+async def get_config():
+    """
+    Get current config
+    """
+
+    APP_MODE = os.environ.get('APP_MODE', 'oss')
```

Comment:
Sorry, I realize this is a draft, but I have a dumb quick question: does this work? 😅
I mean, I wasn't aware that this was in env.

Reply:
Ah I'm planning to move it into env; it would help dedup tracking of `config.json` attributes across backend and frontend

Reply:
Ah I see, you're removing config.json from FE. Idk, in the backend we have a meaning of "config" which is an AppConfig and its components. It would be nice if vars in env are tracked there, if they're necessary. It will also mean they're accepted in `config.toml` as an alternative, as well.

Not sure some of these are necessary though... POSTHOG really isn't used, except in FE, for example. 🤔

GITHUB stuff is probably necessary.

Reply:
Hmm that's a good point, I'll think about this for a bit; couple of initial thoughts

1. It would be nice to centralize config details and dispatch them from one place (motivation for storing in backend)
2. I feel as though `CLIENT-ID` shouldn't be stored in AppConfig since it stays fixed (either local vs prod) and the user shouldn't be tweaking it via `config.toml` (perhaps the same for `APP_MODE`?)

Reply:
🤔 Re: APP_MODE, I wonder, `is_local()` from the middleware, does it achieve the same thing?

Reply:
I initially had `is_local` and decided to remove it since I figured it would prevent tunneling (although admittedly I haven't tested it, which I'll get to soon), but maybe we don't need to bother 😅

Another thought - I see that we have two modes (`oss`, `saas`) but we may end up adding additional (e.g `saas-v2`, etc); in this event having an explicit `APP_MODE` var helps distinguish these cases

If we're confident about binary app modes then `is_local` will achieve the same thing (assuming tunneling isn't affected)

Reply:
I don't know how many integrations there can or will be, but if they're not in scope of this PR, why worry about? There will be time for other PRs. 😅 

Please allow me to make a step back for a moment:

We had integrations before and continue to have (e.g. runtimes). We're removing an auth one, and want another auth, right? To be fair, the sheets implementation was a special purpose/time thing, I doubt anyone else but AH used it.

What would be nice IMHO, would be to keep integrations code localized. Really where it's needed and hopefully not elsewhere. IMHO it keeps the likelihood of things breaking lower. 😅 As Li Boxuan was [suggesting](https://github.com/All-Hands-AI/OpenHands/pull/2045#discussion_r1614430873) at some point back in time, if we help humans make bugs, they might prove great at it!

To duplicate config isn't robust IMO - that seems to invite bug making, and I'm not sure it's necessary. If I have a suggestion at this stage (and late in the night 😂), I guess it's rather to keep it simple.

Reply:
I had a quick chat with @rbren; unfortunately we can't use `is_local` primitives as it'll add complications when self hosting OH. The current solution has one config (on route `/api/options/config`), which can be shared between the backend and frontend.

Since these are values we don't want users to tweak, and some contain secrets it makes sense to store it in `ENV`. I think we could, however, centralize the organization of these values similar to how we've done it for `APP_CONFIG`.

```python
class OpenhandsConfig:
    APP_MODE='oss'
    POSTHOG_CLIENT_KEY='sk'
    
    @classmethod
    def set_config(cls):
        """
        Loop through all class variables and override them with environment variables if they exist.
        """
        for attr_name in dir(cls):
            # Skip built-in attributes and methods
            if attr_name.startswith('__') or callable(getattr(cls, attr_name)):
                continue

            # Get current class variable value
            current_value = getattr(cls, attr_name)

            # Check if an environment variable exists for this attribute
            env_value = os.getenv(attr_name)

            if env_value is not None:
                # Override class variable with the environment variable
                setattr(cls, attr_name, env_value)
                print(f"Overridden {attr_name}: {current_value} -> {env_value}")  # Optional debug log
```